game-core/src/main/java/games/strategy/triplea/image/UnitImageFactory.java

```diff
@@ -142,20 +148,55 @@ public Image getImage(final UnitCategory unit) {
       final String baseImageName, final PlayerId id, final ResourceLoader resourceLoader) {
     // URL uses '/' not '\'
     final String fileName = FILE_NAME_BASE + id.getName() + "/" + baseImageName + ".png";
-    final URL url = resourceLoader.getResource(fileName);
+    final String fileName2 = FILE_NAME_BASE + baseImageName + ".png";
+    final URL url = resourceLoader.getResource(fileName, fileName2);
     return Optional.ofNullable(url);
   }
 
-  private Optional<Image> getBaseImage(final String baseImageName, final PlayerId id) {
+  private Optional<Image> getBaseImage(
+      final String baseImageName, final PlayerId id, final UnitType type) {
     final Optional<URL> imageLocation = getBaseImageUrl(baseImageName, id);
     Image image = null;
     if (imageLocation.isPresent()) {
       image = Toolkit.getDefaultToolkit().getImage(getBaseImageUrl(baseImageName, id).get());
       Util.ensureImageLoaded(image);
+      if (!mapData.ignoreColorizingUnit(type.getName())) {
+        if (mapData.getUnitColor(id.getName()).isPresent()) {
+          final Color color = mapData.getUnitColor(id.getName()).get();
+          image = convertToBufferedImage(image);
+          ImageColorizer.applyColor(color, (BufferedImage) image);
+          if (mapData.shouldFlipUnit(id.getName())) {
+            image = flipImage((BufferedImage) image);
+          }
+        } else if (mapData.getUnitHsb(id.getName()).isPresent()) {
+          final int[] hsb = mapData.getUnitHsb(id.getName()).get();
+          image = convertToBufferedImage(image);
+          ImageColorizer.applyHsb(hsb[0], hsb[1], hsb[2], (BufferedImage) image);
+          if (mapData.shouldFlipUnit(id.getName())) {
+            image = flipImage((BufferedImage) image);
+          }
+        }
+      }
```

Comment:
This is the core of the new functionality where it checks the new map properties and applies necessary colorization and flipping to the unit image.